fs.go

```diff
@@ -0,0 +1,115 @@
+package checkup
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+const indexName = "index.json"
+
+// FS is a way to store checkup results on the local filesystem.
+type FS struct {
+	// The path to the directory where check files will be stored.
+	Dir string `json:"dir"`
+	// The URL corresponding to fs.Dir.
+	URL string `json:"url"`
+
+	// Check files older than CheckExpiry will be
+	// deleted on calls to Maintain(). If this is
+	// the zero value, no old check files will be
+	// deleted.
+	CheckExpiry time.Duration `json:"check_expiry,omitempty"`
+}
+
+func (fs FS) readIndex() (index map[string]int64, err error) {
+	f, err := os.Open(filepath.Join(fs.Dir, indexName))
+	if os.IsNotExist(err) {
+		index = map[string]int64{}
+		err = nil
+		return
+	}
+	if err != nil {
+		return
+	}
+	defer f.Close()
+
+	err = json.NewDecoder(f).Decode(&index)
+	return
+}
```

Comment:
As a style preference, I would prefer not to have named returns here, and instead start the function with `index := make(map[string]int64)`. Even though the function is small, I think the explicit returns are a little easier to follow, especially since you have to make the map anyway.